src/hooks/useUsersFilter.ts

```diff
@@ -0,0 +1,62 @@
+import { IUser } from 'interfaces/user';
+import React, { useMemo } from 'react';
+import { getBasePath } from 'utils/formatPath';
+import { createGlobalStateHook } from 'hooks/useGlobalState';
+
+export interface IUsersFilter {
+    query?: string;
+}
+
+export interface IUsersFilterOutput {
+    filtered: IUser[];
+    filter: IUsersFilter;
+    setFilter: React.Dispatch<React.SetStateAction<IUsersFilter>>;
+}
+
+// Store the users filter state globally, and in localStorage.
+// When changing the format of IUsersFilter, change the version as well.
+const useUsersFilterState = createGlobalStateHook<IUsersFilter>(
+    `${getBasePath()}:useUsersFilter:v1`,
+    { query: '' }
+);
+
+export const useUsersFilter = (users: IUser[]): IUsersFilterOutput => {
+    const [filter, setFilter] = useUsersFilterState();
+
+    const filtered = useMemo(() => {
+        return filterUsers(users, filter);
+    }, [users, filter]);
+
+    return {
+        setFilter,
+        filter,
+        filtered,
+    };
+};
+
+const filterUsers = (users: IUser[], filter: IUsersFilter): IUser[] => {
+    return filterUsersByQuery(users, filter);
+};
+
+const filterUsersByQuery = (users: IUser[], filter: IUsersFilter): IUser[] => {
+    if (!filter.query) {
+        return users;
+    }
+
+    // Try to parse the search query as a RegExp.
+    // Return all users if it can't be parsed.
+    try {
+        const regExp = new RegExp(filter.query, 'i');
+        return users.filter(f => filterUserByRegExp(f, regExp));
```

Comment:
What is _f_ in this context? It looks like a `user` to me? Let's change this to `users.filter(user => filterUserByRegExp(user, regExp))`

Reply:
You're right, I copied this from the existing `useFeaturesFilter` hook where `f` (as in feature) made more sense. Changed to `user` as per your suggestion 👍